### PR TITLE
add k8s Gateway APIs

### PR DESCRIPTION
**Pull request by puertomontt**
https://github.com/solo-io/gloo-mesh-enterprise/issues/11356

### CODE REVIEW

File: codegen/k8s_groups.go [hunk 1]
```diff
@@ -178,12 +179,45 @@ func k8sGroups() []model.Group {
 			Module: "sigs.k8s.io/gateway-api",
 			Resources: []model.Resource{
 				{
-					Kind:          "Gateway",
+					Kind: "Gateway",
```
Comment (dmitri-d) [inline codegen/k8s_groups.go:182]: No longer cluster-scoped?

Comment (puertomontt) [reply codegen/k8s_groups.go:182]: never was

Comment (puertomontt) [reply codegen/k8s_groups.go:182]: image

File: codegen/k8s_groups.go [hunk 2]
```diff
@@ -178,12 +179,45 @@ func k8sGroups() []model.Group {
...
 				},
+				{
+					Kind: "HTTPRoute",
+				},
 			},
 			CustomTypesImportPath: "sigs.k8s.io/gateway-api/apis/v1beta1",
 			ApiRoot:               k8sApiRoot,
 		},
+		{
+			GroupVersion: schema.GroupVersion{
```
Comment (dmitri-d) [inline codegen/k8s_groups.go:196]: iirc, alpha-level api are not available by default (are opt-in during install) and are very likely to change. Is there a requirement to use these?

File: codegen/k8s_groups.go (hunk 1 above)
Comment (dmitri-d) [reply codegen/k8s_groups.go:182]: I'm missing something: it looks like it was cluster-scoped before: https://github.com/solo-io/external-apis/blob/master/codegen/k8s_groups.go#L182

**Review by saiskee — APPROVED** [review]

File: codegen/k8s_groups.go (hunk 2 above)
Comment (puertomontt) [reply codegen/k8s_groups.go:196]: @ashleywang1

File: codegen/k8s_groups.go (hunk 1 above)
Comment (puertomontt) [reply codegen/k8s_groups.go:182]: I mean it is not in the k8s api , having it cluster-scope was a mistake on our part

File: codegen/k8s_groups.go (hunk 2 above)
Comment (ashleywang1) [reply codegen/k8s_groups.go:196]: You're right, this did bite us a bit: https://github.com/solo-io/gloo-mesh-enterprise/issues/12726

**Pull request merged**